### ISSUE TITLE
F/A-18C: Comm triangular channel symbol ^ unavailable

### ISSUE THREAD

**Issue by michaelosthege** [DCS-Skunkworks/dcs-bios#575]
**DCS-BIOS version** v0.8.0 (latest)

**DCS-BIOS Control**

`onUfcComm1DisplayChange` and `onUfcComm2DisplayChange`

**Describe the bug**

The triangle symbol that shows when the radios are turned off (?) should also be routed to the `onUfcComm*DisplayChange` handlers.

~However, it looks like [these lines](https://github.com/DCS-Skunkworks/dcs-bios/blame/7b35b54ded6d61bf4097a1ca123748b24e615ed1/Scripts/DCS-BIOS/lib/modules/aircraft_modules/FA-18C_hornet.lua#L148) replace the `^` by a `1` or `2` and after the `padLeft` the resulting message is identical to the channel strings ` 1` or ` 2`.~
Correction: the replacement is just for changing ``"`3"`` → `"13"` or `"~0"`→`"20"`. The `^` in there is just part of the RegEx.

**To Reproduce**

I don't know how to debug these messages.

**Expected behavior**

The triangle symbol should get reported as or `^` or ` ^` which is the most intuitive character for this.

Then one can show it on 16-segment displays :)

**Screenshots**

![image](https://github.com/DCS-Skunkworks/dcs-bios/assets/5894642/bb4e635f-6974-4952-af4a-c227b5c616a6)

**Comment by charliefoxtwo** [comment]
Hey, yeah the iirc the COMM1/2 displays at least _used to_ show a 1 and 2. Do they not show those anymore? Is this still an issue for the `~` character, or does the display now show the triangle character where 1 used to be, but continues to show a 2 in place of `~`?

Just want to make sure we don't break anything with this fix :)

**Comment by michaelosthege** [comment]
I use my own code to control the 16-segment displays via a shift register, so my interaction point is just the `onUfcComm1DisplayChange` callback. I didn't find documentation about which string to expect and figured out most characters from
* [this forum thread](https://forum.dcs.world/topic/197246-dcs-bios-and-oled-f18/)
* and most importantly [the `comm_channel_map` lookup in the code](https://github.com/DCS-Skunkworks/dcs-bios/blame/7b35b54ded6d61bf4097a1ca123748b24e615ed1/Scripts/DCS-BIOS/lib/modules/aircraft_modules/FA-18C_hornet.lua#L116-L141).

However, that lookup doesn't contain the character shown in the screenshot above.

I would print the `newValue` coming through `onUfcComm1DisplayChange`, but this just gets lots in the `connect-serial-port` noise.

**Comment by charliefoxtwo** [comment]
I'm really just curious about the values displayed in-game at this stage. I'll take a look at the actual values that the game is outputting later on, just trying to get some background before jumping in blind.

The lookup is definitely missing something for the triangle character (looks like two characters?) as I don't remember seeing anything like that last time I tested the hornet UFC.

**Comment by michaelosthege** [comment]
Actually, part of my description was wrong:

This 👇 has nothing to do with the triangle thing. (See description.)
```
	s = s:gsub("^`", "1")
	s = s:gsub("^~", "2")
```

It's probably just a matter of the lookup missing whichever symbol comes from DCS when the triangle is shown.

**Comment by charliefoxtwo** [comment]
Hm, I'm not able to reproduce this symbol. You mention you think this occurs when the radios are turned off - could you give me steps to follow in order to get this symbol to display?

**Comment by michaelosthege** [comment]
For the screenshot I jumped into a mission starting hot on the ground (Syria map Airbase defense), and I did not touch any controls.
I don't know if one can even get to this symbol after using the radio 🤔

**Comment by charliefoxtwo** [comment]
Ok, I started the mission you mentioned and was able to reproduce the issue.

It seems like it has something to do with receiving transmissions, but I'm not sure.

At any rate, DCS outputs `d` when the radio is in that triangle state. I've updated the code to output `<>` in those cases, as that seems to be the closest we could get to any sort of triangle (and it's a two-character display).

Here's the PR https://github.com/DCS-Skunkworks/dcs-bios/pull/576
